modules/ui/maproulette_editor.js

```diff
@@ -185,49 +198,32 @@ export function uiMapRouletteEditor(context) {
       .append('div')
       .attr('class', 'detail-section')
       .merge(detailSection);
-
     const osm = context.services.osm;
     if (!osm) return;
 
-    // Add warning if user is not logged in
-    const hasAuth = osm.authenticated();
+    // Check if user is authenticated with OSM
+    const osmAuth = osm.authenticated();
     let authWarning = detailSection.selectAll('.auth-warning')
-      .data(hasAuth ? [] : [0]);
-
-    authWarning.exit()
-      .transition()
-      .duration(200)
-      .style('opacity', 0)
-      .remove();
-
-    let authEnter = authWarning.enter()
-      .insert('div', '.tag-reference-body')
-      .attr('class', 'field-warning auth-warning')
-      .style('opacity', 0);
-
-    authEnter
-      .call(uiIcon('#rapid-icon-alert', 'inline'));
-
-    authEnter
-      .append('span')
-      .text(l10n.t('map_data.layers.maproulette.login'));
-
-    authEnter
-      .append('a')
-      .attr('target', '_blank')
-      .call(uiIcon('#rapid-icon-out-link', 'inline'))
-      .append('span')
-      .text(l10n.t('login'))
-      .on('click.note-login', d3_event => {
-        d3_event.preventDefault();
-        osm.authenticate();
-      });
-
-    authEnter
-      .transition()
-      .duration(200)
-      .style('opacity', 1);
-
+      .data(osmAuth ? [] : [0]);
+    // If not authenticated with OSM, show OSM login prompt
+    if (!osmAuth) {
```

Comment:
I feel like there might be a subtle bug in here because the `if (!osmAuth)` puts a condition on whether `showAuthWarning` gets called... So in situations where the warning should go away, it won't call that code that contains the `selection.exit().….remove()` exit selection.

In general I think if you are doing a d3 data bind, it's important to always run all the d3 stuff related to the data bind - and could be problematic to hide parts of behind in a conditional.

Reply:
You're correct that the D3 data binding and exit selection should always be executed to make sure the warning is properly removed